capa/features/extractors/viv/insn.py

```diff
@@ -120,6 +120,27 @@ def extract_insn_api_features(fh: FunctionHandle, bb, ih: InsnHandle) -> Iterato
                 yield API(name[1:]), ih.address
             return
 
+        if imports == {}:
+            # no imports implies the binary was likely statically linked.
+            # therefore, we try to use the symbol's table to fetch the api names
+            name = f.vw.name_by_va.get(target)
+            if not name:
+                return
```

Comment:
i dont think we should rely on vivisect's naming here, primarily because vivisect can only associate a single name with any location, even if the symtab has multiple names.

when the symtab has multiple names for a location, we should emit each of them verbatim as their own API feature. this way rule authors have the most flexibility and power (do they want to match the weak names? or only the libc names? it's up to them). also, we don't have to invent too much logic about stemming the prefix - we can just use the data as its present. of course, we emit a few more features than before, but i dont think it's too many nor will affect performance much.

creating the mapping from VA -> list of names should probably happen once and then the results should be re-used on each API extraction. we can use the `ctx` to store the cached data structure.

Reply:
Understood. I was reluctant to use the ELF parser since viv was already there, but now I'll modify the code to use the ELF parser instead. Thanks!